app/helpers/jobs_helper.rb

```diff
@@ -1,3 +1,28 @@
 module JobsHelper
 
+	def build_ganglia_link( host, start_seconds, report_type, node_num, size )
+    OODClusters.each do |c|
+      if host.eql?(c.id.to_s)
+        if c.custom_allow?(:ganglia)
+          @ganglia_cluster = OodCluster::Servers::Ganglia.new(c.custom_config(:ganglia))
+          @ganglia_base = @ganglia_cluster.uri.to_s
+          @ganglia_server = @ganglia_cluster.opt_query.fetch(:h, '') % {h: node_num }
+          @ganglia_uri = (@ganglia_base + '&z=' + size + '&cs=' + start_seconds.to_s + '&g=' + report_type+ "&h=" + @ganglia_server).html_safe
+        end
+      end
+    end
+    @ganglia_uri
+  end
```

Comment:
this guard clause is duplicated because you already check if ganglia is supported prior to building the link, so you could just assume the cluster exists and the ganglia config also exists

if you want an escape hatch put at the bottom of the method `rescue nil` i.e.

``ruby
def build_ganglia_link
  // code here to build link
rescue
  nil
end
```